### PR TITLE
Fix kernel_module_disabled remediation template

### PR DESCRIPTION
**Pull request by Vincent056**
The previous template create file as  `/etc/modprobe.d/75-kernel_module_{{{ KERNMODULE }}}_disabled.conf`, and this PR changes it to `/etc/modprobe.d/{{{ KERNMODULE }}}.conf` so that these rules will pass after the remediation.

### CODE REVIEW

**Comment by jhrozek** [comment]
I think this is a nice catch but I wonder if you also see the same strange issue as I do: some rules, for example `kernel_module_udf_disabled` still have the old remediation, even when I build in a container so hopefully no caching should apply. Do you see the same?

```
              <xccdf-1.2:fix system="urn:xccdf:fix:script:kubernetes" id="kernel_module_udf_disabled">---                                                                                                                                     
apiVersion: machineconfiguration.openshift.io/v1                                                                                                                                                                                              
kind: MachineConfig                                                                                                                                                                                                                           
spec:                                                                                                                                                                                                                                         
  config:                                                                                                                                                                                                                                     
    ignition:                                                                                                                                                                                                                                 
      version: 3.1.0                                                                                                                                                                                                                          
    storage:                                                                                                                                                                                                                                  
      files:                                                                                                                                                                                                                                  
      - contents:                                                                                                                                                                                                                             
          source: data:,install%20udf%20/bin/true%0A                                                                                                                                                                                          
        mode: 0644                                                                                                                                                                                                                            
        path: /etc/modprobe.d/75-kernel_module_udf_disabled.conf                                                                                                                                                                              
        overwrite: true                                                                                                                                                                                                                       
</xccdf-1.2:fix>    
```

on the other hand, some rules are rendered exactly as I expect:
```
              <xccdf-1.2:fix system="urn:xccdf:fix:script:kubernetes" id="kernel_module_uvcvideo_disabled" complexity="low" disruption="medium" reboot="true" strategy="disable">---                                                          
apiVersion: machineconfiguration.openshift.io/v1                                                                                                                                                                                              
kind: MachineConfig                                                                                                                                                                                                                           
spec:                                                                                                                                                                                                                                         
  config:                                                                                                                                                                                                                                     
    ignition:                                                                                                                                                                                                                                 
      version: 3.1.0                                                                                                                                                                                                                          
    storage:                                                                                                                                                                                                                                  
      files:                                                                                                                                                                                                                                  
      - contents:                                                                                                                                                                                                                             
          source: data:,install%20uvcvideo%20/bin/true%0Ablacklist%20uvcvideo%0A                                                                                                                                                              
        mode: 0644                                                                                                                                                                                                                            
        path: /etc/modprobe.d/uvcvideo.conf                                                                                                                                                                                                   
        overwrite: true                                                                                                                                                                                                                       
</xccdf-1.2:fix>  
```

**Comment by Vincent056** [comment]
> I think this is a nice catch but I wonder if you also see the same strange issue as I do: some rules, for example `kernel_module_udf_disabled` still have the old remediation, even when I build in a container so hopefully no caching should apply. Do you see the same?
> 
> ```
>               <xccdf-1.2:fix system="urn:xccdf:fix:script:kubernetes" id="kernel_module_udf_disabled">---                                                                                                                                     
> apiVersion: machineconfiguration.openshift.io/v1                                                                                                                                                                                              
> kind: MachineConfig                                                                                                                                                                                                                           
> spec:                                                                                                                                                                                                                                         
>   config:                                                                                                                                                                                                                                     
>     ignition:                                                                                                                                                                                                                                 
>       version: 3.1.0                                                                                                                                                                                                                          
>     storage:                                                                                                                                                                                                                                  
>       files:                                                                                                                                                                                                                                  
>       - contents:                                                                                                                                                                                                                             
>           source: data:,install%20udf%20/bin/true%0A                                                                                                                                                                                          
>         mode: 0644                                                                                                                                                                                                                            
>         path: /etc/modprobe.d/75-kernel_module_udf_disabled.conf                                                                                                                                                                              
>         overwrite: true                                                                                                                                                                                                                       
> </xccdf-1.2:fix>    
> ```
> 
> on the other hand, some rules are rendered exactly as I expect:
> 
> ```
>               <xccdf-1.2:fix system="urn:xccdf:fix:script:kubernetes" id="kernel_module_uvcvideo_disabled" complexity="low" disruption="medium" reboot="true" strategy="disable">---                                                          
> apiVersion: machineconfiguration.openshift.io/v1                                                                                                                                                                                              
> kind: MachineConfig                                                                                                                                                                                                                           
> spec:                                                                                                                                                                                                                                         
>   config:                                                                                                                                                                                                                                     
>     ignition:                                                                                                                                                                                                                                 
>       version: 3.1.0                                                                                                                                                                                                                          
>     storage:                                                                                                                                                                                                                                  
>       files:                                                                                                                                                                                                                                  
>       - contents:                                                                                                                                                                                                                             
>           source: data:,install%20uvcvideo%20/bin/true%0Ablacklist%20uvcvideo%0A                                                                                                                                                              
>         mode: 0644                                                                                                                                                                                                                            
>         path: /etc/modprobe.d/uvcvideo.conf                                                                                                                                                                                                   
>         overwrite: true                                                                                                                                                                                                                       
> </xccdf-1.2:fix>  
> ```

that's a good find, this is due to some of the rules overriding the template remediation.

**Comment by jhrozek** [comment]
Thank you, tested manually.

I'm going to ack this PR but there are two follow-up things we should do:
1) we should have a YAML e2e file with the initial and the expected results so that we don't regress again
2) This failure:
```
     helpers.go:815: E2E-FAILURE: The expected result for the configure_usbguard_auditbackend rule didn't match. Expected 'PASS', Got 'NOT-APPLICABLE' 
```
is not related to this patch, but I've seen it a couple of times recently. If you agree, I think we should remove the e2e YAML file to unblock the CI, but at the same time we should try to fix it (at least let's file a card and schedule it right away to this sprint)

**Review by jhrozek — APPROVED** [review]
/lgtm

File: shared/templates/kernel_module_disabled/kubernetes.template [hunk 1]
```diff
@@ -15,5 +15,5 @@ spec:
       - contents:
           source: data:,install%20{{{ KERNMODULE | urlencode }}}%20/bin/true%0Ablacklist%20{{{ KERNMODULE | urlencode }}}%0A
         mode: 0644
-        path: /etc/modprobe.d/75-kernel_module_{{{ KERNMODULE }}}_disabled.conf
+        path: /etc/modprobe.d/{{{ KERNMODULE }}}.conf
         overwrite: true
```
Comment (marcusburghardt) [inline shared/templates/kernel_module_disabled/kubernetes.template:19]: Please, insert an empty line at the end.

**Comment by Vincent056** [comment]
The PR https://github.com/ComplianceAsCode/content/pull/9392 should set the expected e2e result of configure_usbguard_auditbackend to `Not-Applicable`, going to add an another commit to address the issue here

**Review by marcusburghardt — APPROVED** [review]

**Comment by hafe** [comment]
What I as a user don't understand is why the check is done against remediation files instead checking the actual kernel modules loaded?

**Comment by Vincent056** [comment]
> What I as a user don't understand is why the check is done against remediation files instead checking the actual kernel modules loaded?

The check was correct, what was wrong was the remediation for those rules

**Comment by hafe** [comment]
Ok so you mean that the remediation did not work? I did not look that closely at loaded modules

**Comment by Vincent056** [comment]
> Ok so you mean that the remediation did not work? I did not look that closely at loaded modules

Yes the issues was caused by remediation

**Pull request merged**